node/src/accountant/scanners/pending_payable_scanner/tx_receipt_interpreter.rs

```diff
@@ -0,0 +1,732 @@
+use crate::accountant::db_access_objects::sent_payable_dao::{
+    Detection, RetrieveCondition, SentPayableDao, SentTx, TxStatus,
+};
+use crate::accountant::db_access_objects::utils::from_unix_timestamp;
+use crate::accountant::scanners::pending_payable_scanner::utils::{
+    FailedValidation, FailedValidationByTable, NormalTxConfirmation, ReceiptScanReport, TxByTable,
+    TxCaseToBeInterpreted, TxReclaim,
+};
+use crate::accountant::scanners::pending_payable_scanner::PendingPayableScanner;
+use crate::blockchain::blockchain_interface::data_structures::{
+    BlockchainTxFailure, StatusReadFromReceiptCheck, TxBlock, TxReceiptError, TxReceiptResult,
+};
+use crate::blockchain::errors::internal_errors::InternalErrorKind;
+use crate::blockchain::errors::BlockchainErrorKind;
+use masq_lib::logger::Logger;
+use std::time::SystemTime;
+use thousands::Separable;
+
+#[derive(Default)]
+pub struct TxReceiptInterpreter {}
+
+impl TxReceiptInterpreter {
+    pub fn compose_receipt_scan_report(
+        &self,
+        tx_cases: Vec<TxCaseToBeInterpreted>,
+        pending_payable_scanner: &PendingPayableScanner,
+        logger: &Logger,
+    ) -> ReceiptScanReport {
+        let scan_report = ReceiptScanReport::default();
+        tx_cases
+            .into_iter()
+            .fold(scan_report, |scan_report_so_far, tx_case| {
+                match tx_case.tx_receipt_result {
+                    TxReceiptResult(Ok(sent_tx_with_status)) => match sent_tx_with_status.status {
+                        StatusReadFromReceiptCheck::Succeeded(tx_block) => {
+                            Self::handle_tx_confirmation(
+                                scan_report_so_far,
+                                tx_case.tx_by_table,
+                                tx_block,
+                                logger,
+                            )
+                        }
+                        StatusReadFromReceiptCheck::Failed(reason) => {
+                            Self::handle_status_with_failure(
+                                scan_report_so_far,
+                                tx_case.tx_by_table,
+                                reason,
+                                logger,
+                            )
+                        }
+                        StatusReadFromReceiptCheck::Pending => Self::handle_still_pending_tx(
+                            scan_report_so_far,
+                            tx_case.tx_by_table,
+                            &*pending_payable_scanner.sent_payable_dao,
+                            logger,
+                        ),
+                    },
+                    TxReceiptResult(Err(e)) => {
+                        Self::handle_rpc_failure(scan_report_so_far, tx_case.tx_by_table, e, logger)
+                    }
+                }
+            })
+    }
+
+    fn handle_still_pending_tx(
+        mut scan_report: ReceiptScanReport,
```

Comment:
An idea is to do it like this:

```rust
impl ReceiptScanReport {
    pub fn handle_still_pending_tx(
        &mut self,
        tx: TxByTable,
        sent_payable_dao: &dyn SentPayableDao,
        logger: &Logger,
    ) {
        // mutate self instead of consuming/returning
    }
}
```